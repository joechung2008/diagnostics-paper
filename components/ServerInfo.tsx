import { Text, View } from "react-native";
import { useServerInfoStyles } from "../hooks/useServerInfoStyles";
import type { ServerInfoProps } from "../lib/types";

const ServerInfo = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}: ServerInfoProps) => {
  const dynamicStyles = useServerInfoStyles();

  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <View style={dynamicStyles.container}>
      <View style={dynamicStyles.table}>
        <View style={dynamicStyles.header}>
          <Text style={dynamicStyles.headerText}>Name</Text>
          <Text style={dynamicStyles.headerText}>Value</Text>
        </View>
        {items.map((item) => (
          <View key={item.name} style={dynamicStyles.row}>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.name}
            </Text>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.value}
            </Text>
          </View>
        ))}
      </View>
    </View>
  );
};

export default ServerInfo;
