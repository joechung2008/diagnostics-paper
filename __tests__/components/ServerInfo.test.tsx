import React from "react";
import { PaperProvider, MD3LightTheme } from "react-native-paper";
import { render } from "@testing-library/react-native";
import ServerInfo from "../../components/ServerInfo";

describe("ServerInfo", () => {
  it("renders correctly with full server info", () => {
    const mockServerInfo = {
      deploymentId: "test-deployment-123",
      extensionSync: {
        totalSyncAllCount: 42,
      },
      hostname: "test-server.example.com",
      nodeVersions: "v18.17.0",
      serverId: "server-456",
      uptime: 1234567890,
    };

    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <ServerInfo {...mockServerInfo} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with minimal server info", () => {
    const mockServerInfo = {
      deploymentId: "minimal-deployment",
      extensionSync: {
        totalSyncAllCount: 0,
      },
      hostname: "minimal-server",
      serverId: "minimal-server-id",
    };

    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <ServerInfo {...mockServerInfo} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
