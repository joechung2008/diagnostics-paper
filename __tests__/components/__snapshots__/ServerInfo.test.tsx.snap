// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ServerInfo renders correctly with full server info 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFBFE",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#FFFBFE",
        "borderColor": "#79747E",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E7E0EC",
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Hostname
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        test-server.example.com
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Uptime
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        1234567890
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Server ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        server-456
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Deployment ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        test-deployment-123
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Node Versions
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        v18.17.0
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Extension Sync | Total Sync All Count
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        42
      </Text>
    </View>
  </View>
</View>
`;

exports[`ServerInfo renders correctly with minimal server info 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFBFE",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#FFFBFE",
        "borderColor": "#79747E",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E7E0EC",
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Hostname
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        minimal-server
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Uptime
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Server ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        minimal-server-id
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Deployment ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        minimal-deployment
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Node Versions
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Extension Sync | Total Sync All Count
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;
